cpp/src/gandiva/engine.cc

```diff
@@ -133,13 +132,17 @@ Status Engine::Make(const std::shared_ptr<Configuration>& conf,
   // ExecutionEngine but only for the lifetime of the builder. Found by
   // inspecting LLVM sources.
   std::string builder_error;
-  std::unique_ptr<llvm::ExecutionEngine> exec_engine{
-      llvm::EngineBuilder(std::move(module))
-          .setMCPU(llvm::sys::getHostCPUName())
-          .setEngineKind(llvm::EngineKind::JIT)
-          .setOptLevel(opt_level)
-          .setErrorStr(&builder_error)
-          .create()};
+
+  llvm::EngineBuilder engine_builder(std::move(module));
+
+  engine_builder.setEngineKind(llvm::EngineKind::JIT)
+      .setOptLevel(opt_level)
+      .setErrorStr(&builder_error);
+
+  if (conf->target_host_cpu()) {
+    engine_builder.setMCPU(llvm::sys::getHostCPUName());
```

Comment:
What is the impact of not setting this? Does LLVM generate sub-optimal code? or does it internally figure out the CPU and generate appropriate code for that?

Reply:
This change conflicts with the other change as the call to getHostCPUName() is done once in the other change

Reply:
It won't use the cpu specific features (eg avx instructions)

Reply:
The conflict is mentioned in the PR description. One needs to be rebased when the other is merged.